src/components/Data/Documents/DocumentListViewColumn.vue

```diff
@@ -0,0 +1,293 @@
+                @click="editDocument(doc.id)"
+                >
+                Edit
+              </a>
+            </li>
+            <li class="divider"></li>
+            <li>
+              <a
+                v-bind:class="{'disabled': !canDelete}"
+                v-title="{active: !canDelete, title: 'You are not allowed to delete this document'}"
+                @click="deleteDocument(doc.id)"
+                >
+                Delete
+              </a>
+            </li>
+          </dropdown>
+        </td>
+        <td class="DocumentColumnItem">
+          {{doc.id}}
+        </td>
+        <td v-for="(attr, k) in customFields" :key="k" class="DocumentColumnItem">
+          <span v-if="parseDocument(attr, doc).isObject" class="relative">
+            <a href="#" @click.prevent="toggleJsonFormatter(attr+doc.id)">{{parseDocument(attr, doc).value}} ...</a>
+            <pre :ref="attr+doc.id" class="DocumentListViewColumn-jsonFormatter" style="visibility: hidden;" v-json-formatter="{content: parseDocument(attr, doc).realValue, open: true}"></pre>
+          </span>
+          <span v-else>
+            {{parseDocument(attr, doc).value}}
+          </span>
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script>
+import Dropdown from '../../Materialize/Dropdown'
+import Title from '../../../directives/title.directive'
+import JsonFormatter from '../../../directives/json-formatter.directive'
+import {
+  canEditDocument,
+  canDeleteDocument
+} from '../../../services/userAuthorization'
+import Autocomplete from '../../Common/Autocomplete'
+import _ from 'lodash'
+
+export default {
+  name: 'DocumentListViewColumn',
+  props: {
+    documents: Array,
+    index: String,
+    collection: String,
+    value: [Object, String, Array],
+    mapping: Object
+  },
+  directives: {
+    Title,
+    JsonFormatter
+  },
+  components: {
+    Dropdown,
+    Autocomplete
+  },
+  mounted () {
+    const columnsConfig = JSON.parse(localStorage.getItem('columnViewConfig') || '{}')
+
+    if (columnsConfig[this.index] && columnsConfig[this.index][this.collection]) {
+      this.customFields = columnsConfig[this.index][this.collection]
+    }
+
+    this.mappingArray = this.buildAttributeList(this.mapping)
+    for (const attr of this.customFields) {
+      this.mappingArray.splice(this.mappingArray.indexOf(attr), 1)
+    }
+  },
+  data() {
+    return {
+      customFields: [],
+      newCustomField: null,
+      mappingArray: []
+    }
+  },
+  methods: {
+    toggleJsonFormatter(id) {
+      if (this.$refs[id][0].style.visibility === 'hidden') {
+        this.$refs[id][0].style.visibility = 'visible'
+      } else {
+        this.$refs[id][0].style.visibility = 'hidden'
+      }
+    },
+    parseDocument(attr, doc) {
+      let ret = attr.includes('.') ? this.getNestedField(doc.content, attr) : doc.content[attr]
+
+      if (typeof ret === 'object') {
```

Comment:
Can ret be null? Because if it can, then a `TypeError` will occur because the code will try to run `parseDocument` on null